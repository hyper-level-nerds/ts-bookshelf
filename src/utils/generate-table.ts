import markdownTable from "markdown-table";

import { FieldData } from "@utils/types";

export function generateTableFromField(field: FieldData) {
    const rows: string[][] = [];

    rows.push(["Name", "Description"]);
    rows.push(["Type", field.type.name]);
    rows.push(["Nullable", field.userData.nullable ? "✔️ Yes" : "❌ No"]);

    if (field.userData.description) {
        rows.push(["Description", field.userData.description]);
    }

    return markdownTable(rows);
}
