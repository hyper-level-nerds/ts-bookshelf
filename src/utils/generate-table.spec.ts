import { generateTableFromField } from "./generate-table";

describe("generateTableFromField() Function", () => {
    it("should be defined", () => expect(generateTableFromField).toBeDefined());
    it("should be a function", () => expect(generateTableFromField).toBeInstanceOf(Function));

    it("should generate a table correctly", () => {
        expect([
            generateTableFromField({
                fieldName: "test",
                type: String,
                userData: {
                    description: "test",
                    nullable: false,
                },
            }),
            generateTableFromField({
                fieldName: "test",
                type: Number,
                userData: {
                    description: "test",
                    nullable: true,
                },
            }),
        ]).toMatchSnapshot();
    });
});
