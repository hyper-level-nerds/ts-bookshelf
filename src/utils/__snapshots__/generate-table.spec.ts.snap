// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTableFromField() Function should generate a table correctly 1`] = `
[
  "| Name        | Description |
| ----------- | ----------- |
| Type        | String      |
| Nullable    | ❌ No        |
| Description | test        |",
  "| Name        | Description |
| ----------- | ----------- |
| Type        | Number      |
| Nullable    | ✔️ Yes      |
| Description | test        |",
]
`;
